kintoclient/tests/test_client.py

```diff
@@ -0,0 +1,541 @@
+        self.assertEquals(session.server_url, mock.sentinel.server_url)
+
+    @mock.patch('kintoclient.requests')
+    def test_no_auth_is_used_by_default(self, requests_mock):
+        response = mock.MagicMock()
+        response.status_code = 200
+        requests_mock.request.return_value = response
+        session = Session('https://example.org')
+        self.assertEquals(session.auth, None)
+        session.request('get', 'https://example.org/test')
+        requests_mock.request.assert_called_with(
+            'get', 'https://example.org/test',
+            data=json.dumps({'data': {}}),
+            headers={'Content-Type': 'application/json'})
+
+    @mock.patch('kintoclient.requests')
+    def test_bad_http_status_raises_exception(self, requests_mock):
+        response = mock.MagicMock()
+        response.status_code = 400
+        requests_mock.request.return_value = response
+        session = Session('https://example.org')
+
+        self.assertRaises(KintoException, session.request, 'get',
+                          'https://example.org/test')
+
+    @mock.patch('kintoclient.requests')
+    def test_session_injects_auth_on_requests(self, requests_mock):
+        response = mock.MagicMock()
+        response.status_code = 200
+        requests_mock.request.return_value = response
+        session = Session(auth=mock.sentinel.auth,
+                          server_url='https://example.org')
+        session.request('get', '/test')
+        requests_mock.request.assert_called_with(
+            'get', 'https://example.org/test',
+            auth=mock.sentinel.auth,
+            data='{"data": {}}',
+            headers={'Content-Type': 'application/json'})
+
+    @mock.patch('kintoclient.requests')
+    def test_requests_arguments_are_forwarded(self, requests_mock):
+        response = mock.MagicMock()
+        response.status_code = 200
+        requests_mock.request.return_value = response
+        session = Session('https://example.org')
+        session.request('get', 'https://example.org/test',
+                        foo=mock.sentinel.bar)
+        requests_mock.request.assert_called_with(
+            'get', 'https://example.org/test',
+            foo=mock.sentinel.bar,
+            data='{"data": {}}',
+            headers={'Content-Type': 'application/json'})
+
+    @mock.patch('kintoclient.requests')
+    def test_passed_data_is_encoded_to_json(self, requests_mock):
+        response = mock.MagicMock()
+        response.status_code = 200
+        requests_mock.request.return_value = response
+        session = Session('https://example.org')
+        session.request('get', 'https://example.org/test',
+                        data={'foo': 'bar'})
+        requests_mock.request.assert_called_with(
+            'get', 'https://example.org/test',
+            data=json.dumps({'data': {'foo': 'bar'}}),
+            headers={'Content-Type': 'application/json'})
+
+    @mock.patch('kintoclient.requests')
```

Comment:
since this is used so much you could also put it in setUp:
````
def setUp(self):
    p = mock.patch('kintoclient.requests')
    self.requests_mock = p.start()
    self.addCleanup(p.stop)
````